pysensors/utils/_constraints.py

```diff
@@ -64,47 +66,350 @@ def get_constrained_sensors_indices_linear(x_min, x_max, y_min, y_max,df):
+            x-coordinate of the center of circle
+        center_y : float,
+            y-coordinate of the center of circle
+        radius : float,
+            radius of the circle
+        '''
+        self.center_x = center_x
+        self.center_y = center_y
+        self.radius = radius
+        
+    def draw(self,ax):
+        c = patches.Circle((self.center_x, self.center_y), self.radius, fill = False, color = 'r', lw = 2)
+        ax.add_patch(c)
+        ax.autoscale_view()
+        
+        
+    def constraint_function(self,x_all_unc, y_all_unc):
+        '''
+        To be Filled
+        '''
+        return ((x_all_unc-self.center_x)**2 + (y_all_unc-self.center_y)**2) - self.radius**2
+        
+           
+class Line(BaseConstraint):
+    '''
+    To be filled
+    '''
+    def __init__(self,x1,x2,y1,y2,**kwgs):
+        super().__init__(**kwgs)
+        '''
+        Attributes
+        ----------
+        x1 : float,
+            x-coordinate of one end-point of the line
+        x2 : float,
+            x-coordinate of the other end-point of the line
+        y1 : float,
+            y-coordinate of one end-point of the line
+        y2 : float,
+            y-coordinate of the other end-point of the line
+        '''
+        self.x1 = x1
+        self.x2 = x2
+        self.y1 = y1
+        self.y2 = y2
+        
+    def draw(self,ax):
+        '''
+        To be Filled
+        '''
+        ax.plot([self.x1,self.x2],[self.y1,self.y2],'-r')
+            
+    def constraint_function(self,x_all_unc, y_all_unc):
+        '''
+        To be Filled
+        '''
+        return (y_all_unc-self.y1)*(self.x2-self.x1) - (self.y2-self.y1)*(x_all_unc-self.x1)
+    
+        
+class Parabola(BaseConstraint):
+    '''
+    Fill in
+    '''
+    def __init__(self,h,k,a, **kwgs):
+        super().__init__(**kwgs)
+        '''
+        Attributes
+        ----------
+        h : float,
+            x-coordinate of the vertex of the parabola we want to be constrained
+        k : float,
+            y-coordinate of the vertex of the parabola we want to be constrained
+        a : float,
+            x-coordinate of the focus of the parabola
+        '''
+        self.h = h
+        self.k = k
+        self.a = a
+        
+class UserDefinedConstraints(BaseConstraint):
+    '''
+    General class for dealing with any form of user defined constraints.
+    Plotting, computing constraints functionalities included. 
+    '''
+    def __init__(self,all_sensors,info,const_path, **kwgs):
+        super().__init__(**kwgs)
+        '''
+        Attributes
+        ----------
+        const_path : 'string',
+            path of the file that constains the user-defined constraint function
+        '''
+        self.all_sensors = all_sensors
+        self.info = info
+        self.const_path = const_path
+        
+    def draw(self,ax):
+        '''
+        To be filled
+        '''
+        nConstraints = len([self.const_path])
```

Comment:
Um, why are we taking the length of a string and using that as the number of constraints?